packages/rsocket-core/src/RSocketMachine.js

```diff
@@ -823,15 +824,13 @@ class RSocketMachineImpl<D, M> implements RSocketMachine<D, M> {
           //critically, if n is 0 now, that's okay because we eagerly decremented it
           if (firstRequest && n >= 0) {
             firstRequest = false;
-            //release the initial frame we received in frame form due to map operator
-            subscriber.onNext(frame);
+            //release the initial payload we received in frame form due to map operator
+            subscriber.onNext(payload);
           }
         },
       });
     }, MAX_REQUEST_N);
-    const framesToPayloads = new FlowableProcessor(payloads, frame =>
-      this._deserializePayload(frame),
-    );
+    const framesToPayloads = new FlowableProcessor(payloads);
     this._receivers.set(streamId, framesToPayloads);
```

Comment:
Bug: `this._receivers` is of type `Map<number, ISubject<Payload<D, M>>>`. The subject is already deserialized payload. Its deserialized and published on `_handleStreamFrame()`. so there is no need to deserialize it again here.